### PR TITLE
[REVIEW] Renaming to bdb_tools

### PR DESCRIPTION
**Pull request by randerzander**
Continuing the work of #175

### CODE REVIEW

File: tpcx_bb/cluster_configuration/cluster-startup.sh [hunk 1]
```diff
@@ -10,21 +10,22 @@ else
 fi
 
 MAX_SYSTEM_MEMORY=$(free -m | awk '/^Mem:/{print $2}')M
-DEVICE_MEMORY_LIMIT="25GB"
+DEVICE_MEMORY_LIMIT="15GB"
```
Comment (VibhuJawa) [inline tpcx_bb/cluster_configuration/cluster-startup.sh:13]: Was this change intentional?

File: tpcx_bb/cluster_configuration/cluster-startup.sh [hunk 2]
```diff
@@ -55,7 +56,9 @@ SCHEDULER_PORT=8786
 DASHBOARD_ADDRESS=8787
 
 if [ "$ROLE" = "SCHEDULER" ]; then
+  echo "I'm scheduler!"
   if [ "$CLUSTER_MODE" = "NVLINK" ]; then
+     echo "Starting sched.."
```
Comment (VibhuJawa) [inline tpcx_bb/cluster_configuration/cluster-startup.sh:61]: We should add this for both `NVLINK` and `TCP` with something like:
```suggestion
     echo "Starting NVLINK based Scheduler"
```

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Thanks for taking on this massive refactoring.  The changes look good to me apart from some questions around the `cluster_startup.sh` script

**Review by beckernick — CHANGES_REQUESTED** [review]
This looks fantastic. Thank you for taking on this work.

Two additional requests:
- The top-level module namespace should be gpu-bdb
- The default performance tracking information should be gpu-bdb in `benchmark_runner/benchmark_config.yaml`

File: tpcx_bb/cluster_configuration/cluster-startup.sh (hunk 1 above)
Comment (randerzander) [reply tpcx_bb/cluster_configuration/cluster-startup.sh:13]: Yes, based on the idea of using the new Dask JIT spilling w/ 32 GB cards.

I'm not sure though that it should be the default setting..

Comment (VibhuJawa) [reply tpcx_bb/cluster_configuration/cluster-startup.sh:13]: We initially wrote this for a dgx-2 with 16 32B cards. Unsure what this setting is for. That said I think we should focus on stability for the defaults than performance so a lower default looks fine. 

CC: @beckernick  for thoughts.

**Comment by randerzander** [comment]
>     * The top-level module namespace should be gpu-bdb

You caught me. Was procrastinating the gitfu to handle the many other files underneath =)

Coming up shortly.

File: tpcx_bb/cluster_configuration/cluster-startup.sh (hunk 1 above)
Comment (beckernick) [reply tpcx_bb/cluster_configuration/cluster-startup.sh:13]: 15GB is the recommendation for running SF1K on a DGX-1. I would recommend we keep the existing default, but don't feel strongly.

File: gpu_bdb/benchmark_runner/benchmark_config.yaml [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 split_row_groups: False
 repartition_small_table: True
 benchmark_runner_include_bsql: False
 
-scheduler_file_path:
+scheduler_file_path: 
 dask_profile: False
 dask_dir: ./
-num_workers: 16
+num_workers: 8
```
Comment (beckernick) [inline gpu_bdb/benchmark_runner/benchmark_config.yaml:14]: If we go back to the original memory limit, this should return to 16. If not, reject this suggestion,
```suggestion
num_workers: 16
```

**Review by beckernick — APPROVED** [review]
Approved. Couple of optional suggestions depending on opinion.

File: gpu_bdb/benchmark_runner/benchmark_config.yaml (hunk 3 above)
Comment (randerzander) [reply gpu_bdb/benchmark_runner/benchmark_config.yaml:14]: @ChrisJar what was the memory limit used w/ 80GB A100s?

Comment (ChrisJar) [reply gpu_bdb/benchmark_runner/benchmark_config.yaml:14]: I used a pool size of 80GB and a device memory limit of 40GB
